ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -445,17 +443,34 @@ private CompactionType checkForCompaction(final CompactionInfo ci,
               ci.getFullPartitionName(), exception);
         }
       }
-      return compactionType;
+    }
+    if (compactionType != null) {
+      return createCompactionRequest(ci, compactionType);
+    } else {
+      return null;
     }
   }
+  
+  private CompactionRequest createCompactionRequest(CompactionCandidate ci, CompactionType type) {
```

Comment:
i don't like this refactor, since it doesn't create a complete CompactionRequest and does some initialization afterward. in `requestCompaction ` you can see the complete request

Reply:
Removed all CompactionInfo class structure related changes